### PR TITLE
[WebHost] Add search filter and collapse button to Supported Games page

### PR DESCRIPTION
**Pull request by LegendaryLinux**
## What is this fixing or adding?
This change adds a search input and collapse buttons to the Supported Games page.

All games begin collapsed, but expand when they are shown as the result of a search. Clearing the search box displays all games in a collapsed state.

## How was this tested?
Written and tested locally on Windows 10 in Chrome and Firefox.

## If this makes graphical changes, please attach screenshots.
https://github.com/ArchipelagoMW/Archipelago/assets/6321333/383ff8d7-fefe-4c91-abb1-99777e553ca4

### CODE REVIEW

File: WebHostLib/static/assets/supportedGames.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      // If the game name includes the search string, display the game. If not, hide it
+      if (gameName.toLowerCase().includes(evt.target.value.toLowerCase())) {
+        header.style.display = null;
+        gameArrow.innerText = '▼';
+        gameInfo.classList.remove('collapsed');
+      } else {
+        console.log(header);
+        header.style.display = 'none';
+        gameArrow.innerText = '▶';
+        gameInfo.classList.add('collapsed');
```
Comment (KonoTyran) [inline WebHostLib/static/assets/supportedGames.js:45]: why are we changing to collapsed state when filtering at all? both here and above.

**Review by KonoTyran — COMMENTED** [review]
Id like to see a collapse/un-collapse all button at the top. i feel like those would be useful to have.

File: WebHostLib/static/assets/supportedGames.js (hunk 1 above)
Comment (KonoTyran) [reply WebHostLib/static/assets/supportedGames.js:45]: ohh this does end up collapsing all when you clear the search bar.
I guess that works well then. this should be fine.

Comment (LegendaryLinux) [reply WebHostLib/static/assets/supportedGames.js:45]: I can bet behind a collapse/un-collapse all button.

**Comment by LegendaryLinux** [comment]
@KonoTyran Added those "Expand All" and "Collapse All" buttons per your request. They respect the currently visible games, as well.

**Review by KonoTyran — APPROVED** [review]
All looks good to me.

**Pull request merged**